bin/control.bat

```diff
@@ -84,110 +84,69 @@ goto checkIgniteHome2
-::
 call "%SCRIPTS_HOME%\include\setenv.bat"
-call "%SCRIPTS_HOME%\include\build-classpath.bat"
 set CP=%IGNITE_LIBS%;%IGNITE_HOME%\libs\optional\ignite-zookeeper\*
 
-::
 :: Process 'restart'.
-::
 set RANDOM_NUMBER_COMMAND="!JAVA_HOME!\bin\java.exe" -cp %CP% org.apache.ignite.startup.cmdline.CommandLineRandomNumberGenerator
 for /f "usebackq tokens=*" %%i in (`!RANDOM_NUMBER_COMMAND!`) do set RANDOM_NUMBER=%%i
 
 set RESTART_SUCCESS_FILE="%IGNITE_HOME%\work\ignite_success_%RANDOM_NUMBER%"
 set RESTART_SUCCESS_OPT=-DIGNITE_SUCCESS_FILE=%RESTART_SUCCESS_FILE%
 
-::
-:: JVM options. See http://java.sun.com/javase/technologies/hotspot/vmoptions.jsp for more details.
-::
-:: ADD YOUR/CHANGE ADDITIONAL OPTIONS HERE
-::
-"%JAVA_HOME%\bin\java.exe" -version 2>&1 | findstr "1\.[7]\." > nul
-if %ERRORLEVEL% equ 0 (
-    if "%CONTROL_JVM_OPTS%" == "" set CONTROL_JVM_OPTS=-Xms256m -Xmx1g
-) else (
-    if "%CONTROL_JVM_OPTS%" == "" set CONTROL_JVM_OPTS=-Xms256m -Xmx1g
-)
+:: Determine Java version function
+:determineJavaVersion
+for /f "tokens=1,2 delims=." %%a in ("%JAVA_VER_STR%.x") do set JAVA_VERSION=%%a
+
+:: Set JVM options with dynamic version check
+call :getJavaSpecificOpts %JAVA_VERSION% "%CONTROL_JVM_OPTS%"
+
+:: Enable assertions if set.
+if "%ENABLE_ASSERTIONS%" == "1" set CONTROL_JVM_OPTS=%CONTROL_JVM_OPTS% -ea
+
+:: Set main class to start service (grid node by default).
+if "%MAIN_CLASS%" == "" set MAIN_CLASS=org.apache.ignite.internal.commandline.CommandHandler
 
-::
 :: Uncomment to enable experimental commands [--wal]
-::
 :: set CONTROL_JVM_OPTS=%CONTROL_JVM_OPTS% -DIGNITE_ENABLE_EXPERIMENTAL_COMMAND=true
 
-::
 :: Uncomment the following GC settings if you see spikes in your throughput due to Garbage Collection.
-::
 :: set CONTROL_JVM_OPTS=%CONTROL_JVM_OPTS% -XX:+UseG1GC
 
-::
 :: Uncomment if you get StackOverflowError.
 :: On 64 bit systems this value can be larger, e.g. -Xss16m
-::
 :: set CONTROL_JVM_OPTS=%CONTROL_JVM_OPTS% -Xss4m
 
-::
 :: Uncomment to set preference to IPv4 stack.
-::
 :: set CONTROL_JVM_OPTS=%CONTROL_JVM_OPTS% -Djava.net.preferIPv4Stack=true
 
-::
-:: Assertions are disabled by default since version 3.5.
-:: If you want to enable them - set 'ENABLE_ASSERTIONS' flag to '1'.
-::
-set ENABLE_ASSERTIONS=1
-
-::
-:: Set '-ea' options if assertions are enabled.
-::
-if %ENABLE_ASSERTIONS% == 1 set CONTROL_JVM_OPTS=%CONTROL_JVM_OPTS% -ea
-
-:run_java
-
-::
-:: Set main class to start service (grid node by default).
-::
-
-if "%MAIN_CLASS%" == "" set MAIN_CLASS=org.apache.ignite.internal.commandline.CommandHandler
-
-::
-:: Remote debugging (JPDA).
-:: Uncomment and change if remote debugging is required.
-:: set CONTROL_JVM_OPTS=-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=8787 %CONTROL_JVM_OPTS%
-::
-
-::
 :: Final CONTROL_JVM_OPTS for Java 9+ compatibility
-::
-call "%SCRIPTS_HOME%\include\jvmdefaults.bat" %MAJOR_JAVA_VER% "%CONTROL_JVM_OPTS%" CONTROL_JVM_OPTS
+if %JAVA_VERSION% GEQ 9 (
```

Comment:
So, your idea is removing `jvmdefaults` call and adding specific options directly in this script, right?
Could you please clarify the exact option(s) required for the control script and not listed in the `jvmdefaults`?
Perhaps, we can add one or two lines to the `jvmdefaults` instead of copying the entire list to this script.